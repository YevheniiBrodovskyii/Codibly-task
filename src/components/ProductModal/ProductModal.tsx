import React, { FC, useEffect, useState } from 'react'
import { IProduct } from '../Product/product.interface'

import CloseIcon from '@mui/icons-material/Close'

import {
	ModalContainer,
	ModalIcon,
	ModalWrapper,
	ModalDescription,
	ModalName,
	ModalYear,
	ModalId,
	ModalColor,
	ModalOverlay,
	ModalPantone,
} from './styled'

const ProductModal: FC<{
	product: IProduct
	modal: boolean
	setModal: React.Dispatch<React.SetStateAction<boolean>>
}> = ({
	product: { id, name, year, color, pantone_value },
	modal,
	setModal,
}) => {
	const [mounted, setMounted] = useState(false)

	useEffect(() => {
		if (modal == true) {
			setMounted(true)
		}
	}, [modal])

	return (
		<>
			<ModalOverlay mounted={mounted} onClick={() => setModal(false)} />
			<ModalContainer mounted={mounted}>
				<ModalIcon onClick={() => setModal(false)}>
					<CloseIcon />
				</ModalIcon>
				<ModalWrapper>
					<ModalDescription>
						<ModalName color={color}>{name}</ModalName>
						<ModalYear>{year}</ModalYear>
						<ModalId>Product id: {id}</ModalId>
					</ModalDescription>
					<ModalColor color={color} />
				</ModalWrapper>
				<ModalPantone>{pantone_value}</ModalPantone>
			</ModalContainer>
		</>
	)
}

export default ProductModal
