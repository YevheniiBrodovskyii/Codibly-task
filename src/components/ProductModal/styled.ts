import styled from 'styled-components'

export const ModalContainer = styled.div<{ mounted: boolean }>`
	position: fixed;
	display: flex;
	flex-direction: column;
	justify-content: space-around;
	background-color: black;
	visibility: ${({ mounted }) => (mounted ? 'visible' : 'hidden')};
	opacity: ${({ mounted }) => (mounted ? '1' : '0')};
	width: calc(40% - 40px);
	height: calc(500px - 20px);
	padding: 10px 20px;
	top: 20%;
	left: 30%;
	border-radius: 10px;
	transition: 0.3s all;
	z-index: 9999;
`
export const ModalIcon = styled.div`
	position: absolute;
	top: 10px;
	left: 10px;
	width: 24px;
	height: 24px;
	color: red;
	cursor: pointer;
`

export const ModalWrapper = styled.div`
	display: flex;
	justify-content: space-between;
`

export const ModalDescription = styled.div`
	width: 55%;
	height: 300px;
	display: flex;
	flex-direction: column;
	justify-content: space-between;
`

export const ModalOverlay = styled.div<{ mounted: boolean }>`
	position: fixed;
	visibility: ${({ mounted }) => (mounted ? 'visible' : 'hidden')};
	opacity: ${({ mounted }) => (mounted ? '0.4' : '0')};
	background-color: black;
	top: 0;
	left: 0;
	width: 100%;
	height: 100%;
	transition: 0.3s all;
	z-index: 8899;
`

export const ModalName = styled.h1`
	color: ${({ color }) => (color ? color : 'white')};
	text-decoration: underline;
	font-size: 60px;
	font-weight: 600;
	margin: 0;
`
export const ModalYear = styled.h2`
	color: white;
	font-size: 40 px;
	font-weight: 300;
	margin: 0;
`

export const ModalId = styled.h3`
	width: 300px;
	color: white;
	border-radius: 10px;
	padding: 5px;
	margin: 0;
	font-size: 35px;
	font-weight: 200;
`

export const ModalColor = styled.div`
	width: 40%;
	height: 100%;
	background-color: ${({ color }) => (color ? color : 'white')};
`
export const ModalPantone = styled.p`
	text-align: right;
	color: white;
	font-size: 20px;
	font-weight: 300;
	margin: 0;
`
