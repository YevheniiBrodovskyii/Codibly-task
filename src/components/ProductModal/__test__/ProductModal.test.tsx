import ProductModal from '../ProductModal'
import { IProduct } from '../../Product/product.interface'
import { RenderResult } from '@testing-library/react'
import { renderWithProviders } from '../../../utils/test-utils'

describe('Main tests', () => {
	let component: RenderResult
	const props: IProduct = {
		id: 1,
		name: 'Test',
		year: 2023,
		color: '#000000',
		pantone_value: '20-2023',
	}
	const setModal = jest.fn()
	beforeEach(() => {
		component = renderWithProviders(
			<ProductModal product={props} modal={true} setModal={setModal} />
		)
	})

	it('should match snapshot on initial render', () => {
		expect(component).toMatchSnapshot()
	})

	// it('should trigger fetch arrow button', async () => {

	// })
})
