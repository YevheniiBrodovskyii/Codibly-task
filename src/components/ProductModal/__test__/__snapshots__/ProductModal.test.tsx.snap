// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main tests should match snapshot on initial render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-eDvSVe iCWvry"
      />
      <div
        class="sc-bcXHqe enNWYc"
      >
        <div
          class="sc-gswNZR eNrUwv"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          class="sc-dkrFOg dhxzHb"
        >
          <div
            class="sc-hLBbgP kKREqP"
          >
            <h1
              class="sc-jSUZER hQovIt"
              color="#000000"
            >
              Test
            </h1>
            <h2
              class="sc-gKPRtg hnBATD"
            >
              2023
            </h2>
            <h3
              class="sc-iBYQkv dhnlck"
            >
              Product id: 
              1
            </h3>
          </div>
          <div
            class="sc-ftTHYK UzabQ"
            color="#000000"
          />
        </div>
        <p
          class="sc-pyfCe FWTta"
        >
          20-2023
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-eDvSVe iCWvry"
    />
    <div
      class="sc-bcXHqe enNWYc"
    >
      <div
        class="sc-gswNZR eNrUwv"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </div>
      <div
        class="sc-dkrFOg dhxzHb"
      >
        <div
          class="sc-hLBbgP kKREqP"
        >
          <h1
            class="sc-jSUZER hQovIt"
            color="#000000"
          >
            Test
          </h1>
          <h2
            class="sc-gKPRtg hnBATD"
          >
            2023
          </h2>
          <h3
            class="sc-iBYQkv dhnlck"
          >
            Product id: 
            1
          </h3>
        </div>
        <div
          class="sc-ftTHYK UzabQ"
          color="#000000"
        />
      </div>
      <p
        class="sc-pyfCe FWTta"
      >
        20-2023
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
